### PR TITLE
annotate embed code for nullability

### PR DESCRIPTION
**Pull request by akiraveliara**
i wanted to do messages too, but that probably would've caused a lot of merge conflicts with caching, cc @Plerx2493

### CODE REVIEW

**Comment by Plerx2493** [comment]
Well i already have over 15 files with merge conflicts. I think I will rebase my branch anyways

**Comment by akiraveliara** [comment]
should i add to this PR or make a separate one, then

**Comment by Plerx2493** [comment]
Add it here if you want

**Review by Plerx2493 — COMMENTED** [review]
LGTM, if the following is true:
- `[]` is equivalent to `Array.Empty<T>()`
- `[^1]` behaves as `.LastOrDefault()`

**Comment by OoLunar** [comment]
> LGTM, if the following is true:
> 
>     * `[]` is equivalent to `Array.Empty<T>()`
> 
>     * `[^1]` behaves as `.LastOrDefault()`

- `[]` is a more efficient version of `Array.Empty` but is functionally the same.
- `[^1]` will grab the last element within the collection. I'm unsure if it'll throw if the array is empty.

**Comment by Plerx2493** [comment]
> > LGTM, if the following is true:
> > ```
> > * `[]` is equivalent to `Array.Empty<T>()`
> > 
> > * `[^1]` behaves as `.LastOrDefault()`
> > ```
> 
>     * `[]` is a more efficient version of `Array.Empty` but is functionally the same.

Took a look at SharpLab and behind the scenes it always uses `Array.Empty<T>` so its ony syntactical sugar

**Comment by Plerx2493** [comment]
image
`[^1]` will throw when used on an empty collection

File: DSharpPlus/Entities/Channel/Message/DiscordMentions.cs [hunk 1]
```diff
@@ -20,19 +20,19 @@ internal class DiscordMentions
...
 
     /// <summary>
     /// The values to be parsed
     /// </summary>
     [JsonProperty("parse", NullValueHandling = NullValueHandling.Ignore)]
-    public IEnumerable<string> Parse { get; }
+    public IEnumerable<string>? Parse { get; }
 
     // WHY IS THERE NO DOCSTRING HERE
     [JsonProperty("replied_user", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (InFTord) [inline DSharpPlus/Entities/Channel/Message/DiscordMentions.cs:38]: сфт цу рфму ф вщсыекштп руку, фы ф пшае

Comment (InFTord) [reply DSharpPlus/Entities/Channel/Message/DiscordMentions.cs:38]: FUCK
*ahem* can we have a docstring here, as a treat

Comment (akiraveliara) [reply DSharpPlus/Entities/Channel/Message/DiscordMentions.cs:38]: no

Comment (InFTord) [reply DSharpPlus/Entities/Channel/Message/DiscordMentions.cs:38]: 😭

Comment (Plerx2493) [reply DSharpPlus/Entities/Channel/Message/DiscordMentions.cs:38]: > 😭

Free pr :when:

**Review by Plerx2493 — APPROVED** [review]

**Pull request merged**